pkg/policy/api/rule_validation.go

```diff
@@ -43,6 +44,12 @@ func (r Rule) Sanitize() error {
 		}
 	}
 
+	for _, v := range r.IngressVisibility {
+		if err := v.sanitize(); err != nil {
```

Comment:
Won't this run `sanitize()` on a copy of the `IngressVisibilityRule` so the following in `sanitize()` will have no effect:
```
v.L7Protocol, err = ParseL7ParserType(string(v.L7Protocol))
```